src/Tribe/Shortcode/Event_Details.php

```diff
@@ -0,0 +1,214 @@
+<?php
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+/**
+ * Setup the Event Details Shortcode to be able to place the Details for a Event on other pages
+ *
+ * @since 4.1
+ */
+class Tribe__Events__Shortcode__Event_Details {
+
+	/**
+	 * We use a Private constructor to Prevent `new Class` usage
+	 * Correct usage: `Tribe__Events__Shortcode__Event_Details::instance()`
+	 *
+	 * @since  4.1
+	 * @return  void
+	 */
+	private function __construct() {
+
+	}
+
+	/**
+	 * Static Singleton Factory Method
+	 *
+	 * @since  4.1
+	 * @return Tribe__Events__Shortcode__Event_Details
+	 */
+	public static function instance() {
+		static $instance;
+
+		if ( ! $instance ) {
+			$instance = new self;
+		}
+
+		return $instance;
+	}
+
+	/**
+	 * Add the necessary hooks as the correct moment in WordPress
+	 *
+	 * @since  4.1
+	 * @return  void
+	 */
+	public static function hook() {
+		$myself = self::instance();
+
+		add_action( 'init', array( $myself, 'add_shortcode' ) );
+	}
+
+	/**
+	 * Static private variable holding this Shortcode Namespace, which should always be "tribe"
+	 *
+	 * @since 4.1
+	 * @var string
+	 */
+	private $nspace = 'tribe';
+
+	/**
+	 * Static private variable holding this Shortcode Slug
+	 *
+	 * @since 4.1
+	 * @var string
+	 */
+	private $slug = 'event-details';
+
+	/**
+	 * Creates the Shortcode tag based on the static variables $nspace and $slug
+	 *
+	 * @since 4.1
+	 * @return string
+	 */
+	public function get_shortcode_tag() {
+		/**
+		 * Use this hook to filter the tribe namespace on shortcodes
+		 *
+		 * @since 4.1
+		 * @var string
+		 *
+		 * @param string Static Class Name
+		 * @param object Instance of this class
+		 */
+		$nspace = apply_filters( 'tribe_events_shortcode_namespace', $this->nspace, __CLASS__, $this );
```

Comment:
I didn't over engineer it now, but I want to create as Abstract (on common) for this as soon as we plan on a second Shortcode. We could include this as a feature for 4.2, what do you think?